config/config-api/src/com/thoughtworks/go/config/materials/MaterialConfigs.java

```diff
@@ -289,12 +318,15 @@ public void setConfigAttributes(Object attributes) {
             addMaterialConfig(getTfsMaterial(), (Map) attributeMap.get(TfsMaterialConfig.TYPE));
         } else if (PackageMaterialConfig.TYPE.equals(materialType)) {
             addMaterialConfig(getPackageMaterial(), (Map) attributeMap.get(PackageMaterialConfig.TYPE));
+        } else if (PluggableSCMMaterialConfig.TYPE.equals(materialType)) {
+            addMaterialConfig(getSCMMaterial(), (Map) attributeMap.get(PluggableSCMMaterialConfig.TYPE));
         }
     }
 
     public boolean scmMaterialsHaveDestination() {
-        for (ScmMaterialConfig scmMaterial : filterScmMaterials()) {
-            if (!scmMaterial.hasDestination()) {
+        for (MaterialConfig scmMaterial : getSCMAndPluggableSCMConfigs()) {
+            String destination = (scmMaterial instanceof ScmMaterialConfig) ? ((ScmMaterialConfig) scmMaterial).getFolder() : ((PluggableSCMMaterialConfig) scmMaterial).getFolder();
```

Comment:
The MaterialConfig interface has a getFolder() method. Why is this needed? The condition check, I mean.